Spigot-API-Patches/0132-Add-hand-to-bucket-events.patch

```diff
@@ -0,0 +1,106 @@
+From 33752df84404b098c3756be93e9cc7cdf0eeca3d Mon Sep 17 00:00:00 2001
+From: BillyGalbreath <Blake.Galbreath@GMail.com>
+Date: Thu, 2 Aug 2018 08:44:20 -0500
+Subject: [PATCH] Add hand to bucket events
+
+
+diff --git a/src/main/java/org/bukkit/event/player/PlayerBucketEmptyEvent.java b/src/main/java/org/bukkit/event/player/PlayerBucketEmptyEvent.java
+index 8fb121a9..4af9882c 100644
+--- a/src/main/java/org/bukkit/event/player/PlayerBucketEmptyEvent.java
++++ b/src/main/java/org/bukkit/event/player/PlayerBucketEmptyEvent.java
+@@ -5,6 +5,7 @@ import org.bukkit.block.Block;
+ import org.bukkit.block.BlockFace;
+ import org.bukkit.entity.Player;
+ import org.bukkit.event.HandlerList;
++import org.bukkit.inventory.EquipmentSlot;
+ import org.bukkit.inventory.ItemStack;
+ 
+ /**
+@@ -13,8 +14,10 @@ import org.bukkit.inventory.ItemStack;
+ public class PlayerBucketEmptyEvent extends PlayerBucketEvent {
+     private static final HandlerList handlers = new HandlerList();
+ 
+-    public PlayerBucketEmptyEvent(final Player who, final Block blockClicked, final BlockFace blockFace, final Material bucket, final ItemStack itemInHand) {
+-        super(who, blockClicked, blockFace, bucket, itemInHand);
++    // Paper start - add EquipmentSlot
++    public PlayerBucketEmptyEvent(final Player who, final Block blockClicked, final BlockFace blockFace, final Material bucket, final ItemStack itemInHand, final EquipmentSlot hand) {
++        super(who, blockClicked, blockFace, bucket, itemInHand, hand);
++        // Paper end
+     }
+ 
+     @Override
+diff --git a/src/main/java/org/bukkit/event/player/PlayerBucketEvent.java b/src/main/java/org/bukkit/event/player/PlayerBucketEvent.java
+index 56584687..236d6762 100644
+--- a/src/main/java/org/bukkit/event/player/PlayerBucketEvent.java
++++ b/src/main/java/org/bukkit/event/player/PlayerBucketEvent.java
+@@ -5,6 +5,7 @@ import org.bukkit.block.Block;
+ import org.bukkit.block.BlockFace;
+ import org.bukkit.entity.Player;
+ import org.bukkit.event.Cancellable;
++import org.bukkit.inventory.EquipmentSlot;
+ import org.bukkit.inventory.ItemStack;
+ 
+ /**
+@@ -16,13 +17,15 @@ public abstract class PlayerBucketEvent extends PlayerEvent implements Cancellab
+     private final Block blockClicked;
+     private final BlockFace blockFace;
+     private final Material bucket;
++    private final EquipmentSlot hand; // Paper
+ 
+-    public PlayerBucketEvent(final Player who, final Block blockClicked, final BlockFace blockFace, final Material bucket, final ItemStack itemInHand) {
++    public PlayerBucketEvent(final Player who, final Block blockClicked, final BlockFace blockFace, final Material bucket, final ItemStack itemInHand, final EquipmentSlot hand ) { // Paper - add EquipmentSlot
+         super(who);
+         this.blockClicked = blockClicked;
+         this.blockFace = blockFace;
+         this.itemStack = itemInHand;
+         this.bucket = bucket;
++        this.hand = hand; // Paper
```

Comment:
while it seems silly, we generally have to maintain old constructors still.

Plugins may manually call events to 'test conditions' themselves, so this can be a breaking change (as with the other ctor changes)

pass null in the original ctors, then at this location, if null, check if item in main type == bucket use main, else use offhand

Reply:
Not silly ^_^ However, I dont understand your last sentence. Pushed updated code.

Reply:
Not silly ^_^ However, I dont understand your last sentence. Pushed updated code.

Reply:
He means to make it so that if null, it will attempt to determine the hand that is in use by checking the players main hand for the item